From 0fb2897026b69e48f65bcd0ff3d7baceb14e6490 Mon Sep 17 00:00:00 2001
From: hyphop <art@khadas.com>
Date: Tue, 28 Apr 2020 01:17:42 +0900
Subject: [PATCH 21/38] boot_seq_change

1) change boot seq to => spi usb0 mmc0 mmc1 mmc2 pxe dhcp
2) disable usb net
3) add spiflash bootcmd
---
 include/config_distro_bootcmd.h | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/include/config_distro_bootcmd.h b/include/config_distro_bootcmd.h
index d2ef89704c..df195663cd 100644
--- a/include/config_distro_bootcmd.h
+++ b/include/config_distro_bootcmd.h
@@ -299,7 +299,7 @@
 #ifdef CONFIG_CMD_USB
 #define BOOTENV_RUN_NET_USB_START "run boot_net_usb_start; "
 #define BOOTENV_SHARED_USB \
-	"boot_net_usb_start=usb start\0" \
+	"boot_net_usb_start=;\0" \
 	"usb_boot=" \
 		"usb start; " \
 		BOOTENV_SHARED_BLKDEV_BODY(usb)
@@ -474,7 +474,8 @@
 #define BOOTENV_DEV_NAME(devtypeu, devtypel, instance, ...) \
 	BOOTENV_DEV_NAME_##devtypeu(devtypeu, devtypel, instance, ## __VA_ARGS__)
 #define BOOTENV_BOOT_TARGETS \
-	"boot_targets=" BOOT_TARGET_DEVICES(BOOTENV_DEV_NAME) "\0"
+	"bootcmd_spi=test \"$boot_source\" = \"spi\" && sf probe && sf read $loadaddr 0x160000 0x008000 && script - - 0 1\0" \
+	"boot_targets=spi usb0 " BOOT_TARGET_MMC(BOOTENV_DEV_NAME) " pxe dhcp \0"
 
 #define BOOTENV_DEV(devtypeu, devtypel, instance, ...) \
 	BOOTENV_DEV_##devtypeu(devtypeu, devtypel, instance, ## __VA_ARGS__)
-- 
2.17.1

